.github/workflows/test.yml

```diff
@@ -666,108 +665,45 @@ jobs:
 
     steps:
-      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
-
-      - uses: actions/setup-java@8df1039502a15bceb9433410b1a100fbe190c53b # v4
-        with:
-          distribution: "temurin"
-          java-version: "17"
-      - name: Setup Nextflow
-        uses: nf-core/setup-nextflow@v2
-
-      - name: Install nf-test
-        uses: nf-core/setup-nf-test@v1
+      - name: Run nf-test Action
+        uses: ./.github/actions/nf-test-action
         with:
-          version: ${{ env.NFTEST_VER }}
-
-      - name: Setup apptainer
-        if: matrix.profile == 'singularity'
-        uses: eWaterCycle/setup-apptainer@main
+          path: ${{ matrix.path }}
+          profile: ${{ matrix.profile }}
 
-      - name: Set up Singularity
-        if: matrix.profile == 'singularity'
-        run: |
-          mkdir -p $NXF_SINGULARITY_CACHEDIR
-          mkdir -p $NXF_SINGULARITY_LIBRARYDIR
-
-      - name: Set up Python
-        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5
-        with:
-          python-version: "3.11"
-
-      - uses: actions/cache@6849a6489940f00c2f30c0fb92c6274307ccb58a # v4
-        id: cache-pip-pdiff
-        with:
-          path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-pdiff
-          restore-keys: |
-            ${{ runner.os }}-pip-pdiff
-
-      - name: Install Python dependencies
-        run: python -m pip install --upgrade pip pdiff cryptography
-
-      - name: Set up miniconda
-        if: matrix.profile == 'conda'
-        uses: conda-incubator/setup-miniconda@d2e6a045a86077fb6cad6f5adf368e9076ddaa8d # v3
-        with:
-          miniconda-version: "latest"
-          auto-update-conda: true
-          channels: conda-forge,bioconda
-
-      - name: Conda setup
-        if: matrix.profile == 'conda'
-        run: |
-          conda clean -a
-          conda install -n base conda-libmamba-solver
-          conda config --set solver libmamba
-          echo $(realpath $CONDA)/condabin >> $GITHUB_PATH
-          echo $(realpath python) >> $GITHUB_PATH
-
-      # Set up secrets
-      - name: Set up nextflow secrets
-        # TODO Only run if the tag includes `sentieon`
-        if: env.SENTIEON_ENCRYPTION_KEY != null && env.SENTIEON_LICENSE_MESSAGE != null
-        run: |
-          nextflow secrets set SENTIEON_AUTH_DATA $(python3 tests/modules/nf-core/sentieon/license_message.py encrypt --key "${{ secrets.SENTIEON_ENCRYPTION_KEY }}" --message "${{ secrets.SENTIEON_LICENSE_MESSAGE }}")
+  nf-test-gpu:
+    runs-on: "gpu"
+    name: nf-test-gpu
+    needs: [nf-test-changes]
+    if: ( needs.nf-test-changes.outputs.paths != '[]' )
+    strategy:
+      fail-fast: false
+      matrix:
+        path: ["${{ fromJson(needs.nf-test-changes.outputs.paths) }}"]
+        profile: [conda, docker_self_hosted, singularity]
+        include:
+          - path: modules/nf-core/parabricks/applybqsr
```

Comment:
Will this trigger tests for these three modules whenever `if: ( needs.nf-test-changes.outputs.paths != '[]' )` is true.